src/transformers/models/data2vec/modeling_data2vec_audio.py

```diff
@@ -138,9 +138,7 @@ def compute_num_masked_span(input_length):
 
     # compute number of masked spans in batch
     input_lengths = (
-        attention_mask.sum(-1).detach().tolist()
-        if attention_mask is not None
-        else [sequence_length for _ in range(batch_size)]
+        attention_mask.sum(-1).detach() if attention_mask is not None else [sequence_length for _ in range(batch_size)]
```

Comment:
tolist inquires a device synch + is not supported by compile I think @sanchit-gandhi